test/integration/targets/dnf/tasks/dnf.yml

```diff
@@ -581,6 +581,41 @@
     that:
       - "not dnf_result is changed"
 
+- name: Install latest httpd
+  dnf:
+    name: httpd
+    state: latest
+  register: dnf_result
+
+- name: downgrade httpd package 
+  command: dnf downgrade httpd -y
+  register: httpd_downgrade
+
+- name: check httpd exists only
+  dnf:
+    name: httpd
+    state: present
+  register: httpd_present
+
+- name: check httpd exists only
+  dnf:
+    name: httpd
+    state: installed
+  register: httpd_installed
+ 
+- name: uninstall httpd
+  dnf:
+    name: httpd
+    state: removed
+
+- name: Verify installed/present state is not changed
```

Comment:
These assertions are passing on devel for me, so I don't think the bug is getting tested. The issue https://github.com/ansible/ansible/issues/76463 also seems to be running into the problem in check mode, so we should make sure check mode gives the same 'changed' result as non-check mode.

Reply:
https://github.com/ansible/ansible/issues/76463#issuecomment-988340340

Reply:
Didn't mean we should replace these, just that we should *also* be testing check mode.